inmap_test.go

```diff
@@ -580,7 +580,14 @@ func TestConverge(t *testing.T) {
 			t.Logf("%s completed after %d iterations.", convergenceNames[i], iterations)
 		}
 
-		r, err := d.Results(false, true, map[string]string{"PrimPM25": "PrimaryPM25"})
+		o := NewOutputter("", false, map[string]string{"PrimPM25": "PrimaryPM25"}, nil)
+
+		err := o.CheckForDerivatives()
+		if err != nil {
+			t.Error(err)
+		}
+
+		r, err := d.Results(o.AllLayers, o.OutputVariables, o.GetVariables, o.OutputFunctions)
```

Comment:
Yes, although I think there may be some misunderstanding about what it is we're talking about. Here's an example of something similar: https://github.com/spatialmodel/inmap/blob/master/io.go#L71

and another one:
https://github.com/spatialmodel/inmap/blob/master/io.go#L280

Basically when you use a struct (or a reference [i.e., "*"] to a struct) as an argument to function, then you can access the struct fields and methods within the function. There isn't any rule that says you have to access all of the fields.